Meshtastic/Helpers/BLEManager.swift

```diff
@@ -874,11 +881,26 @@ class BLEManager: NSObject, CBPeripheralDelegate, MqttClientProxyManagerDelegate
 								if decodedInfo.packet.rxTime > 0 {
 									hopNode?.lastHeard = Date(timeIntervalSince1970: TimeInterval(Int64(decodedInfo.packet.rxTime)))
 								}
-								hopNodes.append(traceRouteHop)
 							}
+							hopNodes.append(traceRouteHop)
 							routeString += "\(hopNode?.user?.longName ?? (node == 4294967295 ? "Repeater" : String(hopNode?.num.toHex() ?? "unknown".localized))) \(hopNode?.viaMqtt ?? false ? "MQTT" : "") (\(traceRouteHop.snr > 0 ? hopNode?.snr ?? 0.0 : 0.0)dB) --> "
 						}
-						var routeBackString = traceRoute?.node?.user?.longName ?? "unknown".localized
+						let destinationHop = TraceRouteHopEntity(context: context)
+						destinationHop.name = traceRoute?.node?.user?.longName ?? "unknown".localized
+						destinationHop.time = Date()
+						destinationHop.snr = Float(routingMessage.snrTowards.last ?? 0 / 4)
+						destinationHop.num = traceRoute?.node?.num ?? 0
+						if let mostRecent = traceRoute?.node?.positions?.lastObject as? PositionEntity, mostRecent.time! >= Calendar.current.date(byAdding: .hour, value: -24, to: Date())! {
+							destinationHop.altitude = mostRecent.altitude
+							destinationHop.latitudeI = mostRecent.latitudeI
+							destinationHop.longitudeI = mostRecent.longitudeI
+							traceRoute?.hasPositions = true
+						}
+						hopNodes.append(destinationHop)
+						/// Add the destination node to the end of the route towards string and the beginning of teh route back string
+						routeString += "\(traceRoute?.node?.user?.longName ?? "unknown".localized) \((traceRoute?.node?.num ?? 0).toHex()) \(traceRoute?.node?.snr ?? 0 > 0 ? traceRoute?.node?.snr ?? 0 : 0.0)dB)"
```

Comment:
I think it would also be better to use the SNR values from the traceroute and not those corresponding to a node, because those will change when subsequent packets are received.